src/api.js/Monitor.js

```diff
@@ -0,0 +1,115 @@
+"use strict";
+
+const { require } = ChromeUtils.import(
+  "resource://devtools/shared/Loader.jsm",
+  {},
+);
+const { DebuggerServer } = require("devtools/server/main");
+const { DebuggerClient } = require("devtools/shared/client/debugger-client");
+const { TargetFactory } = require("devtools/client/framework/target");
+const Services = require("Services");
+const { NetMonitorAPI } = require("devtools/client/netmonitor/src/api");
+const {
+  buildHarLog,
+} = require("devtools/client/netmonitor/src/har/har-builder-utils");
+
+export class Monitor {
+  /**
+   * @param TabBase tabBase
+   */
+  constructor(tabBase) {
+    this.tabBase = tabBase;
+  }
+
+  /**
+   * Create our own debugger client object
+   * @returns {Promise<DebuggerClient>}
+   */
+  async getDebuggerServerClient() {
+    if (this.client) {
+      return this.client;
+    }
+    DebuggerServer.init();
+    DebuggerServer.registerAllActors();
+    const client = new DebuggerClient(DebuggerServer.connectPipe());
+    await client.connect();
+    this.client = client;
+    return this.client;
+  }
+
+  /**
+   * Initialize NetMonitorAPI object and connect to the
+   * Firefox backend for a specific tab
+   * @returns {Promise<void>}
+   */
+  async startTabMonitoring() {
+    const client = await this.getDebuggerServerClient();
+    const tabBase = this.tabBase;
+
+    // Get the debugger's version of the tab object
+    const response = await client.getTab({ tab: tabBase.nativeTab });
+    const form = response.tab;
+
+    // Activate NetMonitor for tab
+    const target = await TargetFactory.forRemoteTab({
+      client,
+      form,
+      chrome: false,
+    });
+    const MockToolbox = {
+      target,
+      getPanel: () => {},
+    };
+    await this.getNetMonitorAPI().connect(MockToolbox);
```

Comment:
It will only start recording from here, when calling `this.getNetMonitorAPI().connect`. 
So you will have to reload the current page or load a new one in the tab in order to get some data out of `NetMonitorAPI.getHar()` (and wait for the end of load).

Reply:
It doesn't seem to work currently. I can reload the page twenty times, write new URLs in the address bar of the monitored tab etc after this statement - without anything but empty HAR contents being returned by NetMonitor afterwards.

Reply:
Hum. I don't know then.

It would be helpful to trace what is happening here:
https://searchfox.org/mozilla-central/source/devtools/client/netmonitor/src/api.js#107
`options.items` should be a non empty array.

Then ensure that this code is trigerred:
https://searchfox.org/mozilla-central/source/devtools/client/netmonitor/src/connector/firefox-connector.js#71
Without the call to `addListeners` here, nothing will be recorded.

Finally, you could also ensure the redux actions are correctly dispatched for each request, from here:
  https://searchfox.org/mozilla-central/source/devtools/client/netmonitor/src/connector/firefox-data-provider.js#76-77
  Otherwise, the `store` won't be populated.

If all that is correct, then I'm really out of clue.